app/main.py

```diff
@@ -1,6 +1,27 @@
-from typing import Callable
+from typing import Callable, Any, Tuple, Dict
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage: Dict[Tuple, Any] = {}
+
+    def wrapper(*args) -> Any:
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[args]
+
+        print("Calculating new result")
+        result = func(*args)
+        cache_storage[args] = result
+        return result
```

Comment:
No need to create result variable here
You'll be able to avoid duplicating return after making the change 🤠

Reply:
Thanks, it seems i fixed it